core-client/src/main/java/org/glassfish/jersey/client/JerseyWebTarget.java

```diff
@@ -149,10 +149,9 @@ public JerseyWebTarget matrixParam(String name, Object... values) throws NullPoi
     public JerseyWebTarget queryParam(String name, Object... values) throws NullPointerException {
         checkNotClosed();
         UriBuilder uriBuilder = getUriBuilder();
-        if (uriBuilder instanceof JerseyUriBuilder) {
-            ((JerseyUriBuilder) uriBuilder).setQueryParamStyle((JerseyQueryParamStyle) this.getConfiguration()
-                            .getProperty(ClientProperties.QUERY_PARAM_STYLE)
-            );
+        Object queryParamProperty = this.getConfiguration().getProperty(ClientProperties.QUERY_PARAM_STYLE);
+        if (queryParamProperty != null && uriBuilder instanceof JerseyUriBuilder) {
+            ((JerseyUriBuilder) uriBuilder).setQueryParamStyle((JerseyQueryParamStyle) queryParamProperty);
```

Comment:
Would it be better if we had a check whether the queryParamProperty is JerseyQueryParamStyle?

Reply:
changed